synapse/config/registration.py

```diff
@@ -44,6 +46,9 @@ def read_config(self, config):
         )
 
         self.auto_join_rooms = config.get("auto_join_rooms", [])
+        for room_alias in self.auto_join_rooms:
+            if not RoomAlias.is_valid(room_alias):
+                raise ConfigError('Invalid auto_join_rooms entry %s' % room_alias)
```

Comment:
generally prefer `% (room_alias, )` in case `room_alias` turns out to be a tuple